src/server/dfly_main.cc

```diff
@@ -96,6 +99,114 @@ namespace dfly {
+  auto const regex = std::regex(reg_match_expr);
+  std::smatch match;
+  if (std::regex_match(from, match, regex) && match.size() > 1) {
+    // the second entry is the match to the group that holds the version string
+    return match[1].str();
+  } else {
+    return std::string{kInvalidVersion};
+  }
+}
+
+std::string GetRemoteVersion(ProactorBase* proactor, SSL_CTX* ssl_context, const std::string host,
+                             std::string_view service, const std::string& resource,
+                             const std::string& ver_header) {
+  namespace bh = boost::beast::http;
+  using ResponseType = bh::response<bh::string_body>;
+
+  bh::request<bh::string_body> req{bh::verb::get, resource, 11 /*http 1.1*/};
+  req.set(bh::field::host, host);
+  req.set(bh::field::user_agent, ver_header);
+  ResponseType res;
+  TlsClient http_client{proactor};
+  http_client.set_connect_timeout_ms(2000);
+
+  auto ec = http_client.Connect(host, service, ssl_context);
+  if (!ec) {
+    ec = http_client.Send(req, &res);
+    if (!ec) {
+      VLOG(1) << "successfully got response from HTTP GET for host " << host << ":" << service
+              << "/" << resource << " response code is " << res.result();
+
+      if (res.result() == bh::status::ok) {
+        return GetVersionString(res.body());
+      }
+    } else {
+      VLOG(1) << "failed to process send HTTP GET to " << host << "@" << service << " at resource '"
+              << resource << "'";
+    }
+  } else {
+    VLOG(1) << "failed to connect: " << ec.message();
+  }
+  return std::string{kInvalidVersion};
+}
+
+struct VersionMonitor {
+  fibers_ext::Fiber version_fiber_;
+  fibers_ext::Done monitor_ver_done_;
+
+  void Run(ProactorPool* proactor_pool) {
+    version_fiber_ = proactor_pool->GetNextProactor()->LaunchFiber([this] { RunTask(); });
+  }
+
+  void Shutdown() {
+    monitor_ver_done_.Notify();
+    if (version_fiber_.IsJoinable()) {
+      version_fiber_.Join();
+    }
+  }
+
+ private:
+  void RunTask();
+};
+
+void VersionMonitor::RunTask() {
+  const auto loop_sleep_time = std::chrono::hours(24);  // every 24 hours
+
+  const std::string host_name = "run-stage-43h632k3xq-ew.a.run.app";
+  const std::string_view port = "443";
+  const std::string resource = "/v1";
+  const std::string dev_version_name = "dev";
+  const std::string version_header = std::string("DragonflyDB/") + kGitTag;
+
+  // Don't run in dev environment - i.e. where we don't build with
+  // real version number
+  if (kGitTag == dev_version_name) {
+    return;
+  }
+
+  SSL_CTX* context = TlsClient::CreateSslContext();
+  if (!context) {
+    VLOG(1) << "failed to create SSL context - cannot run version monitoring";
+    return;
+  }
+
+  while (true) {
+    const std::string remote_version =
+        GetRemoteVersion(ProactorBase::me(), context, host_name, port, resource, version_header);
+
+    if (remote_version != kGitTag) {
+      LOG_EVERY_N(INFO, 1) << "Your current version '" << kGitTag
```

Comment:
it should not be LOG_EVERY_N, it should be LOG_FIRST_N

Reply:
ok